File: src/main/java/com/example/mediaApp/model/entity/Group.java
package com.example.mediaApp.model.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.util.List;

@Entity(name = "Group")
@Table(name = "_group")
@Getter
@Setter
@NoArgsConstructor
@ToString(onlyExplicitlyIncluded = true, includeFieldNames = false)
public class Group {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private String name;

    private String description;

    @Lob
    @Column(length = 1000000)
    private byte[] image;

    @ManyToOne(
            cascade = CascadeType.ALL,
            fetch = FetchType.LAZY
    )
    @JoinColumn(name = "admin_user_id")
    private AppUser adminUser;

    @ManyToMany
    @JoinTable(
            name = "group_participant",
            joinColumns = @JoinColumn(name = "group_id"),
            inverseJoinColumns = @JoinColumn(name = "participant_user_id")
    )
    private List<AppUser> participantUsers;

    @OneToMany(
            mappedBy = "group",
            cascade = CascadeType.ALL,
            orphanRemoval = true
    )
    private List<GroupRequest> groupRequests;

    @OneToMany(
            mappedBy = "group",
            cascade = CascadeType.ALL,
            orphanRemoval = true
    )
    private List<Post> posts;
}

File: src/main/angular-media-app/src/app/models/group.ts
import { GroupRequest } from "./group-request";
import { Post } from "./post";
import { User } from "./user";

export class Group {
    id: number;
    name: string;
    description: string;
    image: number[] | null;
    adminUser: User | null;
    participantUsers: User[];
    groupRequests: GroupRequest[];
    posts: Post[];
    imgSrc: string;

    constructor(){
        this.id = 0;
        this.name = "";
        this.description = "";
        this.image = null;
        this.adminUser = null;
        this.participantUsers = [];
        this.groupRequests = [];
        this.posts = [];
        this.imgSrc = "";
    }

    static convertNewGroup(group: Group): Group{
        let newGroup: Group = new Group();
        newGroup.id = group.id;
        newGroup.name = group.name;
        newGroup.description = group.description;
        newGroup.image = group.image;
        newGroup.adminUser = group.adminUser;
        newGroup.participantUsers = group.participantUsers;
        newGroup.groupRequests = group.groupRequests;
        newGroup.posts = group.posts;
        newGroup.imgSrc = group.getImageSrc();
        return newGroup;
    }

    getImageSrc(): string{
        if(this.image){
            return 'data:image/jpeg;base64,' + this.image;
        }
        return '';
    }
}
